packages/yew/Cargo.toml

```
@@ -29,6 +29,7 @@ thiserror = "1.0"
 
 futures = { version = "0.3", optional = true }
 html-escape = { version = "0.2.9", optional = true }
+implicit-clone = { version = "0.2", features = ["map"] }
```

Do we need to enable `map` feature? I don't think we use it.

---

No... I added it because there was the re-export and it felt like a core feature. But on the other hand, it's an invitation to use an anti-pattern that is not even useful for Yew. It's maybe best to remove it entirely from implicit-clone too

---

I agree that we should remove it:

```suggestion
implicit-clone = { version = "0.2" }
```

---

What do you think about removing it from implicit-clone too? You're also maintaining a UI framework so I guess you probably met a lot of use cases where we need to pass lists in properties and maybe different types.

On my side I didn't find any use of passing maps. The only component that does use a more complex structure for state is the [tree](https://yewprint.rm.rs/#tree) but I opted for using a specialized tree library instead. The ones that use Vec will greatly benefit from IArray.

---

`Attributes` is more-or-less a map, although it's more specialized I guess. Overall, it makes sense to have such a thing in implicit-clone, though I confess I haven't looked at the implementation what you're currently using to implement it.

---

Ah yes I forgot there is that Attributes thingy...

Well... yes probably Attributes is the same (at least same idea) than IMap but it shouldn't be a concern here. Attributes is used for internal machinery in Yew. Here we are talking about API and what tools we want to expose to the user. I don't know if anyone needs Attributes? Do they? Oh... it's public API... haha okay so maybe :grin: why is this public API?? xD (cc @siku2 maybe you know something? I think you've been maintaining Yew the longest)

---

@hamza1311 I forgot to mention this but the map feature brings zero dependencies as it depends on indexmap which yew depends on anyway.

...come to think of it... :thinking: even if I didn't find a use in my use cases, it seems that having an immutable map collection is still something useful. IMap has 2 advantages:
1. it implements ImplicitClone: this means all its API yields clones instead of references which is very useful for properties
2. you can create static collections (for small collections)

I still removed the feature in the dependencies because it's optional anyway. I did it because I originally re-exported but now that we don't re-export there is absolutely no point. (tbh... I really don't get why we ban re-export :woman_shrugging:)

@WorldSEnder I just checked Attributes and it has quite some big differences:
- Attributes can only hold strings vs IMap can hold anything that implements ImplicitClone itself: it could be faster to use Attributes
- Attributes has a variant `Dynamic` which allow using subslices of an existing set of keys/values": it's probably used to optimize Yew somewhere... I'm not sure if IMap should also implement this, I don't see much value tbh.

Overall I have the feeling that Attributes is really specialized for Yew. The reason why it is public API is because it is used by yew-macro so it has to be.

---

@hamza1311 oh nevermind what I said. We have to keep the feature map otherwise I can't make the impls for IntoPropValue which is a yew trait. Chicken and egg problem.

Unless there is a way to put a `#[cfg()]` on the features of a child dependency, I don't see any other way. It's not a big deal anyway since it brings no dependencies.

---

It doesn't matter either way. It's fine that you removed it. Keeping it around wouldn't have mattered either